kfs-core/src/main/java/edu/arizona/kfs/pdp/batch/service/impl/ShippingInvoiceProcessServiceImpl.java

```diff
@@ -0,0 +1,558 @@
+package edu.arizona.kfs.pdp.batch.service.impl;
+
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.coa.service.AccountService;
+import org.kuali.kfs.fp.document.DisbursementVoucherConstants;
+import org.kuali.kfs.pdp.PdpConstants;
+import org.kuali.kfs.pdp.PdpKeyConstants;
+import org.kuali.kfs.pdp.PdpPropertyConstants;
+import org.kuali.kfs.pdp.businessobject.Batch;
+import org.kuali.kfs.pdp.businessobject.CustomerProfile;
+import org.kuali.kfs.pdp.businessobject.PaymentAccountDetail;
+import org.kuali.kfs.pdp.businessobject.PaymentDetail;
+import org.kuali.kfs.pdp.businessobject.PaymentGroup;
+import org.kuali.kfs.pdp.businessobject.PaymentNoteText;
+import org.kuali.kfs.pdp.businessobject.PaymentStatus;
+import org.kuali.kfs.sys.context.SpringContext;
+import org.kuali.kfs.vnd.VendorConstants;
+import org.kuali.kfs.vnd.businessobject.VendorAddress;
+import org.kuali.kfs.vnd.businessobject.VendorDetail;
+import org.kuali.kfs.vnd.businessobject.VendorHeader;
+import org.kuali.kfs.vnd.document.service.VendorService;
+import org.kuali.rice.core.api.datetime.DateTimeService;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+import org.kuali.rice.core.api.util.type.KualiInteger;
+import org.kuali.rice.coreservice.framework.parameter.ParameterService;
+import org.kuali.rice.krad.service.BusinessObjectService;
+import org.kuali.rice.krad.util.ObjectUtils;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.arizona.kfs.pdp.batch.service.ShippingInvoiceProcessService;
+import edu.arizona.kfs.pdp.businessobject.ShippingAccount;
+import edu.arizona.kfs.pdp.businessobject.ShippingInvoice;
+import edu.arizona.kfs.pdp.businessobject.ShippingInvoiceTracking;
+import edu.arizona.kfs.pdp.dataaccess.ShippingInvoiceDao;
+import edu.arizona.kfs.sys.KFSConstants;
+import edu.arizona.kfs.sys.KFSParameterKeyConstants;
+
+/**
+ * This is the default implementation of the ShippingInvoiceProcessService interface.
+ * 
+ * @see edu.arizona.kfs.pdp.batch.service.ShippingInvoiceProcessService
+ */
+@Transactional
+public class ShippingInvoiceProcessServiceImpl implements ShippingInvoiceProcessService {
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ShippingInvoiceProcessServiceImpl.class);
+    
+    protected BusinessObjectService businessObjectService;
+    protected ParameterService parameterService;
+    protected DateTimeService dateTimeService;
+    protected AccountService accountService;
+    protected ShippingInvoiceDao shippingInvoiceDao;
+    
+    protected class KFSAccountingInfo {

```

Comment:
I did some poking around in the KFS 3 code, and it goes back to how we integrated account info with the PaymentDetail object.